web/src/user_group_edit_members.ts

```diff
@@ -36,6 +36,14 @@ function get_potential_members(): User[] {
         if (group.members.has(person.user_id)) {
             return false;
         }
+
+        if (
+            person.user_id === current_user.user_id &&
+            !settings_data.can_join_user_group(current_group_id)
+        ) {
+            return false;
+        }
```

Comment:
I think this block can be dropped.

Reply:
Yes, if we are having the other setting as "Who can add anyone to group" then this can be dropped.